ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java

```diff
@@ -660,6 +645,34 @@ public static Long getLongHeader(Map<String, Object> args, String header, boolea
     return null;
   }
 
+  /**
+   * Gets the value of a header as a {@code boolean}.
+   * @param args a map of arguments to be used to look for {@code header}.
+   * @param header the name of the header.
+   * @param required if {@code true}, {@link RestServiceException} will be thrown if {@code header} is not present
+   *                 in {@code args}.
+   * @return {@code true} if the header's value is {@code "true"} (case-insensitive), or {@code false} if the header's
+   *         value is {@code "false} (case-insensitive) or the header is not present and {@code required} is
+   *         {@code false}.
+   * @throws RestServiceException same as cases of {@link #getHeader(Map, String, boolean)} and if the value cannot be
+   *                              converted to a {@code boolean}.
+   */
+  public static boolean getBooleanHeader(Map<String, Object> args, String header, boolean required)
+      throws RestServiceException {
+    boolean booleanValue;
+    String stringValue = getHeader(args, header, required);
+    if (stringValue == null || "false".equalsIgnoreCase(stringValue)) {
```

Comment:
minor: The `false` and `true` strings should be constants (although I think java handles repeatedly creating the same strings well)

Reply:
this is ok, since the string literal will be stored as a constant by the compiler